sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileSystems.java

```diff
@@ -434,56 +421,55 @@ static FileSystemRegistrar getRegistrarInternal(String scheme) {
   /********************************** METHODS FOR REGISTRATION **********************************/
 
   /**
-   * Loads available {@link FileSystemRegistrar} services.
+   * Sets the default configuration in workers.
+   *
+   * <p>It will be used in {@link FileSystemRegistrar FileSystemRegistrars} for all schemes.
    */
-  private static void loadFileSystemRegistrars() {
-    SCHEME_TO_REGISTRAR.clear();
+  public static void setDefaultConfigInWorkers(PipelineOptions options) {
+    checkNotNull(options, "options");
+    SCHEME_TO_FILESYSTEM.clear();
```

Comment:
we may need locks around this now -- there's a race condition if `setDefaultConfigInWorkers` is called while anything else is happening.

FWICT the Dataflow worker is the only one that initializes once at the beginning.